src/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -385,7 +390,7 @@ private function hasEndOfHeadElement(string $html): bool
      */
     private function removeHtmlComments(string $html): string
     {
-        $result = \preg_replace('/<!--[^-]*+(?:-(?!->)[^-]*+)*+(?:-->|$)/', '', $html);
+        $result = \preg_replace(static::HTML_COMMENT_PATTERN, '', $html);
```

Comment:
Please generally let's use `self::`, not `static:: `, in order to commicate that this constant is not intended to be overwritten in subclasses.

Reply:
This would also fix the Psalm errors.  However, I'd draw your attention to #487 for #228 in which the code was changed throughout to use `static::` rather than `self::` for accessing all constants.

Reply:
Ah, I see. This was before I had understood when to use `self::` and when to use `static::` (both concerning the functional difference as well as what this communicates). :-)

Reply:
OK.  I think `self::` is correct to access class constants - they are, after all, constants (so really not expected to be overridden).